### ISSUE TITLE
total-functions/no-unsafe-readonly-mutable-assignment false positive

### ISSUE THREAD

**Issue by JustFly1984** [danielnixon/eslint-plugin-total-functions#497]
Error in vscode:

`Passing a readonly type to a function that expects a mutable type can lead to unexpected mutation in the readonly value. eslinttotal-functions/no-unsafe-readonly-mutable-assignment`

points to children in if statement.

```typescript
import React, { memo, isValidElement } from 'react'
import type { ReactNode } from 'react'

interface Props {
  children: ReactNode
}

function Component({ children }: Props): JSX.Element | null {
  if (!isValidElement<ReactNode>(children)) return null
  
  return <div>{children}</div>
}
```

**Comment by danielnixon** [comment]
This boils down to:

```ts
const a = "";

// error here
const b: {} = a;
```

The rule is considering the `{}` mutable for some reason, probably incorrectly. I'll fix it.

**Comment by danielnixon** [comment]
This has the same underlying cause as https://github.com/danielnixon/eslint-plugin-total-functions/issues/577

**Comment by danielnixon** [comment]
Fixed in v6.6.1
